pkg/controller/installation/windows_controller.go

```diff
@@ -0,0 +1,597 @@
+	err = utils.AddSecretsWatch(c, "", common.OperatorNamespace())
+	if err != nil {
+		return fmt.Errorf("tigera-windows-controller failed to watch secrets: %w", err)
+	}
+
+	if err = utils.AddConfigMapWatch(c, active.ActiveConfigMapName, common.CalicoNamespace, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("tigera-windows-controller failed to watch ConfigMap %s: %w", active.ActiveConfigMapName, err)
+	}
+
+	// Only watch AmazonCloudIntegration if the CRD is available
+	if ri.amazonCRDExists {
+		err = c.Watch(&source.Kind{Type: &operator.AmazonCloudIntegration{}}, &handler.EnqueueRequestForObject{})
+		if err != nil {
+			logw.V(5).Info("Failed to create AmazonCloudIntegration watch", "err", err)
+			return fmt.Errorf("amazoncloudintegration-controller failed to watch primary resource: %w", err)
+		}
+	}
+
+	if err = imageset.AddImageSetWatch(c); err != nil {
+		return fmt.Errorf("tigera-windows-controller failed to watch ImageSet: %w", err)
+	}
+
+	for _, t := range secondaryResources() {
+		pred := predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				// Create occurs because we've created it, so we can safely ignore it.
+				return false
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				if utils.IgnoreObject(e.ObjectOld) && !utils.IgnoreObject(e.ObjectNew) {
+					// Don't skip the removal of the "ignore" annotation. We want to
+					// reconcile when that happens.
+					return true
+				}
+				// Otherwise, ignore updates to objects when metadata.Generation does not change.
+				return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
+			},
+		}
+		err = c.Watch(&source.Kind{Type: t}, &handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &operator.Installation{},
+		}, pred)
+		if err != nil {
+			return fmt.Errorf("tigera-windows-controller failed to watch %s: %w", t, err)
+		}
+	}
+
+	// Watch for changes to KubeControllersConfiguration.
+	err = c.Watch(&source.Kind{Type: &crdv1.KubeControllersConfiguration{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("tigera-windows-controller failed to watch KubeControllersConfiguration resource: %w", err)
+	}
+
+	// Watch for changes to FelixConfiguration.
+	err = c.Watch(&source.Kind{Type: &crdv1.FelixConfiguration{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("tigera-windows-controller failed to watch FelixConfiguration resource: %w", err)
+	}
+
+	// Watch for changes to BGPConfiguration.
+	err = c.Watch(&source.Kind{Type: &crdv1.BGPConfiguration{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("tigera-windows-controller failed to watch BGPConfiguration resource: %w", err)
+	}
+
+	if ri.enterpriseCRDsExist {
+		// Watch for changes to primary resource ManagementCluster
+		err = c.Watch(&source.Kind{Type: &operator.ManagementCluster{}}, &handler.EnqueueRequestForObject{})
```

Comment:
I don't think this controller actually uses these CRDs:

- ManagementCluster
- ManagementClusterConnection
- ManagerInternalLSSecretName
- PrometheusTLSSecretName (It does use NodePrometheusTLSServerSecret and PrometheusClientTLSSecretName)
- LogCollector

So, I think we can remove the watches for them.